core/services/ocr2/plugins/mercury/config/config.go

```diff
@@ -29,26 +40,64 @@ type PluginConfig struct {
 	NativeFeedID *mercuryutils.FeedID `json:"nativeFeedID" toml:"nativeFeedID"`
 }
 
-func ValidatePluginConfig(config PluginConfig, feedID mercuryutils.FeedID) (merr error) {
-	if config.RawServerURL == "" {
-		merr = errors.New("mercury: ServerURL must be specified")
+func validateURL(rawServerURL string) error {
+	var normalizedURI string
+	if schemeRegexp.MatchString(rawServerURL) {
+		normalizedURI = rawServerURL
 	} else {
-		var normalizedURI string
-		if schemeRegexp.MatchString(config.RawServerURL) {
-			normalizedURI = config.RawServerURL
+		normalizedURI = fmt.Sprintf("wss://%s", rawServerURL)
+	}
+	uri, err := url.ParseRequestURI(normalizedURI)
+	if err != nil {
+		return pkgerrors.Errorf(`Mercury: invalid value for ServerURL, got: %q`, rawServerURL)
+	}
+	if uri.Scheme != "wss" {
+		return pkgerrors.Errorf(`Mercury: invalid scheme specified for MercuryServer, got: %q (scheme: %q) but expected a websocket url e.g. "192.0.2.2:4242" or "wss://192.0.2.2:4242"`, rawServerURL, uri.Scheme)
+	}
+	return nil
+}
+
+type Server struct {
+	URL    string
+	PubKey utils.PlainHexBytes
+}
+
+func (p PluginConfig) GetServers() (servers []Server) {
+	if p.RawServerURL != "" {
+		return []Server{{URL: wssRegexp.ReplaceAllString(p.RawServerURL, ""), PubKey: p.ServerPubKey}}
+	}
+	for url, pubKey := range p.Servers {
+		servers = append(servers, Server{URL: wssRegexp.ReplaceAllString(url, ""), PubKey: pubKey})
+	}
+	sort.Slice(servers, func(i, j int) bool {
+		return servers[i].URL < servers[j].URL
+	})
+	return
+}
+
+func ValidatePluginConfig(config PluginConfig, feedID mercuryutils.FeedID) (merr error) {
```

Comment:
Should we protect against the same Server URL being specified multiple times? I don't think there are any technical sharp edges with allowing it (`Checkout` from the mercury pool will make sure a wsrpc client only has one connection to the mercury server IFIUC)

Reply:
It unmarshals into a map so duplicates would be handled however the TOML library usually handles that. Either the dupe would be ignored, or it would throw an error. Either way its impossible to have dupes after unmarshalling.